src/pages/docs/creators/ray/cluster-yaml.md

```diff
@@ -104,4 +104,88 @@ By default, Ray on Golem uses prepackaged VM images including [relatively fresh]
+
+## Budget control
+
+### Spending hard limit
+
+Within `provider.parameters` section there is the `budget_limit` property.
+It defines the maximum amount of GLMs paid for the whole cluster operations - from `ray up` until `ray down`.
+
+At the moment, when the spending reaches the limit, Ray on Golem will stop spending, effectively terminating the cluster nodes.
+
+```yaml
+provider:
+  parameters:
+
+    # Maximum amount of GLMs that's going to be spent for the whole cluster
+    budget_limit: 2
+```
+
+### Avoiding too-expensive providers
+
+You can use `provider.parameters.node_config.budget_control` section to define the limits on providers' prices.
+Ray on Golem won't work with providers who exceed any of the following price settings.
+
+#### Maximum provider prices
+
+Golem providers charge in three ways. They charge:
+- the initial price at the start of image deployment,
+- CPU usage price for the total time (in seconds) their CPUs spent computing,
+- duration price for the total time (in seconds) they spent up and running,
+
+So for example, if you rent a 3 CPU node for 15 minutes and the average load is 80% you will be charged `initial_price + duration_price * 15*60 + cpu_usage_price * 3 * 0.8 * 15*60`.
+
+The following properties allow you to reject providers with any of the prices exceeding your limits.
+- `max_initial_price`
+- `max_cpu_sec_price`
+- `max_duration_sec_price`
+
+```yaml
+provider:
+  parameters:
+    node_config:
+      budget_control:
+
+        # Amount of GLMs for worker initiation 
+        # causing Golem provider offer to be rejected
+        max_initial_price: 0.5
+
+        # Amount of GLMs for CPU utilization per second 
+        # causing Golem provider offer to be rejected
+        max_cpu_sec_price: 0.0005
+
+        # Amount of GLMs for each second that worker runs 
+        # causing Golem provider offer to be rejected
+        max_duration_sec_price: 0.0005
+```
+
+#### Choosing the cheapest providers - maximum expected usage cost
+
+To work with the cheapest provider, we combine all [three prices](#maximum-provider-prices) into one value.
+
+Estimate the rough time your cluster will be up, and the expected CPU load during that time. Set `expected_duration_minutes` and `expected_cpu_load` properties to compute each provider's expected usage cost.
+
+All providers' offers will be sorted by estimated expected usage cost, allowing Ray on Golem to negotiate with the cheapest nodes first.
+
+You might also use `max_expected_usage_cost` to unconditionally cut off providers with too big an expected usage cost (in GLMs).
+
+```yaml
+provider:
+  parameters:
+    node_config:
+      budget_control:
+
+        # Estimated expected load and duration for worker 
+        # allowing the picking of the least expensive Golem providers' offers first.
+        # If not provided, offers will be picked at random.
+        # Both values need to be defined or undefined together.
+        expected_cpu_load: 0.8
```

Comment:
plus.... really - if we call it `cpu_load` and the value is expressed as a float, I _really_ have no idea whether that means a per-core load or a cumulative load... especially that in unix world, load is given as a sum of all individual core loads...

```suggestion
        expected_cpu_load: 0.8
```